config/metrics.exs

```diff
@@ -0,0 +1,12 @@
+use Mix.Config
+
+config :ex_metrics,
+  metrics: [
```

Comment:
🤔 so you have to list all your metrics up front? Can't we just send whatever metric?

Reply:
yes, so this is what I was going to discuss with people. This is an optional feature.

This was a step towards automating our alarms and dashboards for any app that uses the ExMetrics library. It also opens the door to other potential features:

The pros are:
1. All metrics are clearly defined in a single place

2. Avoid spelling errors in metrics

3. Encourages more discipline in metric naming - poorly named metrics stand out

4. We can make use of the `mix list_metrics` command the ExMetrics library provides

5. We can create more mix commands in the `ExMetrics` library - for example it is possible that using this list of metrics with other defined information, we could generate cloudformation for alarms, and/or grafana dashboards.

6. If in the future we add optional thresholds for metrics, in the metrics config file, we could also specify a callback module that performs an action when a metric reaches the threshold

7. These checks are only done in `test` and `dev` mix envs, on live, a metric with any name is sent through.

It does have cons:
1. We might not want to implement the mix tasks that generate alarms and grafana charts because this is too close to the code, and is currently kept separate for reasons

Reply:
I think the added con is a lack of dynamic metrics, i.e in requester we currently metric the component name and status code, and we don't want to specify in advance a different metric for every combination of every possible component id and status

Reply:
well, my plan was great albeit short-lived

Reply:
actually, it seems it wasn't even great 😆

Reply:
Thinking about this more, I believe I discarded that kind of dynamic metric as we cannot query cloudwatch from grafana with something like "error.component.non200.*.*" so we'd have to hardcode those metrics to grafana & specifying alarms anyway, so we might as well do that here?

Reply:
I've now changed the ExMetrics library so that we only log an error message when a metric is undefined in this config. This means that we can continue to specify the metrics we know about here and wish to use when building alarms and dashboards - but does not limit us in the test and dev environments from using dynamic metrics

https://github.com/bbc/ExMetrics/pull/5